webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/properties/PropertiesRepository.scala

```diff
@@ -128,16 +130,19 @@ class InMemoryPropertiesRepository(
     }
   }
 
-  override def getNodeProps(nodeId: NodeId)(implicit qc: QueryContext): IOResult[Option[Chunk[NodePropertyHierarchy]]] = {
+  override def getNodeProps(nodeId: NodeId)(implicit qc: QueryContext): IOResult[ResolvedNodePropertyHierarchy] = {
     for {
       // needed to check access to node
-      _     <- nodeFactRepository.get(nodeId)
+      _     <- nodeFactRepository
+                 .get(nodeId)
+                 .chainError(s"Node properties not found for node with ID ${nodeId.value} because node does not exist")
       props <- nodeProps.get
     } yield {
-      props.get(nodeId)
+      props.get(nodeId).orNotFound
     }
   }
 
+  // This API inherently swallows known errors, it does only return success node property in the collection
```

Comment:
Okay, I did not pay much to this method because I kept the previous behavior which is that if the cache is empty we just do nothing (and this method is used to get expiration policy, if the node is not found, the default value is used)    

I would even return a `Map[NodeId, Either[NodePropertyError, NodePropertyHierarchy]]`, but I would not know what to do in the case of a real property resolution error, or just when it is not found : rethrow the error in the former case and use the default in the latter ?